<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QWidget" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>784</width>
    <height>676</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>DSMWindow</string>
  </property>
  <layout class="QGridLayout" name="gridLayout" columnstretch="0,5">
   <item row="0" column="0">
    <layout class="QVBoxLayout" name="vl_options">
     <property name="spacing">
      <number>4</number>
     </property>
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>文件操作</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_openFile">
       <property name="text">
        <string>打开文件</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_save">
       <property name="text">
        <string>保存</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>动态分区分配算法</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_FF">
       <property name="text">
        <string>首次适应算法</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="autoExclusive">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_NF">
       <property name="text">
        <string>循环首次适应算法</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="autoExclusive">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_BF">
       <property name="text">
        <string>最佳适应算法</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="autoExclusive">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_WF">
       <property name="text">
        <string>最坏适应算法</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="autoExclusive">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_compact">
       <property name="text">
        <string>支持紧凑算法</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>进程分区操作</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_autoCreateProcess">
       <property name="text">
        <string>自动生成进程</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_addProcess">
       <property name="text">
        <string>添加进程</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_endProcess">
       <property name="text">
        <string>结束所有进程</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_resetMemory">
       <property name="text">
        <string>重置内存分区</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_1">
       <property name="text">
        <string>流程控制</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_startSystem">
       <property name="text">
        <string>开始模拟</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_endSystem">
       <property name="text">
        <string>暂停模拟</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout"/>
     </item>
     <item>
      <spacer name="vs_leftOptions">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QHBoxLayout" name="hl_statistics">
       <property name="topMargin">
        <number>0</number>
       </property>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="topMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="lb_partitionCount">
           <property name="text">
            <string>分区数量：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="lb_processCount">
           <property name="text">
            <string>进程数量：</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QLabel" name="lb_allocated">
           <property name="text">
            <string>已分配内存：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="lb_memorySize">
           <property name="text">
            <string>内存总大小：</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item row="0" column="1">
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <layout class="QVBoxLayout" name="vl_ptable">
       <item>
        <widget class="QLabel" name="lb_ppc">
         <property name="text">
          <string>进程执行情况</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTableWidget" name="tw_processTable">
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="verticalScrollMode">
          <enum>QAbstractItemView::ScrollPerPixel</enum>
         </property>
         <property name="horizontalScrollMode">
          <enum>QAbstractItemView::ScrollPerPixel</enum>
         </property>
         <column>
          <property name="text">
           <string>进程名</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>PID</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>首地址</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>申请大小</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>预计剩余时间</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <layout class="QVBoxLayout" name="vl_mtable">
       <item>
        <widget class="QLabel" name="lb_mac">
         <property name="text">
          <string>内存分配状况</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTableWidget" name="tw_memoryTable">
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="verticalScrollMode">
          <enum>QAbstractItemView::ScrollPerPixel</enum>
         </property>
         <property name="horizontalScrollMode">
          <enum>QAbstractItemView::ScrollPerPixel</enum>
         </property>
         <column>
          <property name="text">
           <string>分区号</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>分区首地址</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>分区大小</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>分配状态</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="8"/>
 <resources/>
 <connections/>
</ui>
